mozilla-services/socorro
# bug-1898341, bug-1898345: add "host" tag, fix metrics key prefix

This adds a "host" tag to emitted metrics when running in GCP. This is derived from "HOSTNAME" if it exists, otherwise it defaults to `socket.gethostname()` like our other services.

This changes Sentry and logging to use `HOSTNAME` configuration variable rather than `HOST_ID`. This brings us in line with other services as we migrate to GCP.

This also adds `"socorro."` prefix to all emitted keys, but only for the GCP environments so we don't make a major change to socorro running in AWS at this time. This brings keys in line with our other services.

In order to do this, I had to create a singleton `METRICS` in `socorro/libmarkus.py` and then rework everything to use that.

To test:

1. run the tests--they have assertions about keys and tags
2. run socorro, process crashes, make sure metrics are emitted and look correct
   * processor, webapp, and crontabber metrics keys **are NOT** prefixed with `socorro`
   * keys **do NOT** include a `host` tag
3. set `CLOUD_PROVIDER=GCP`, run socorro, process crashes, make sure metrics are emitted and look correct
   * processor, webapp, and crontabber metrics keys **are** prefixed with `socorro`
   * keys **do** include a `host` tag
